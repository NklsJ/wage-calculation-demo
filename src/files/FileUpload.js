// @flow
import React from 'react'

type Props = {
  onChange: Function,
  isUploading: boolean,
}

const FileUpload = ({onChange, isUploading}: Props) => {

  return (
    <div>
      <h3>Upload CSV</h3>
      <input
        type="file"
        accept=".csv"
        disabled={isUploading}
        onChange={async event => await onChange(event)}
        multiple/>
    </div>
  )
}

export default FileUpload
