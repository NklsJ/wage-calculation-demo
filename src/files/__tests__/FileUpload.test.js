import React from 'react'
import {shallow} from 'enzyme'

import FileUpload from '../FileUpload'

describe('FileUpload component: ', () => {

  it('Should render correctly', () => {
    const component = shallow(<FileUpload onChange={() => {}} isUploading={false} />)
    expect(component).toMatchSnapshot()
  })

  it('Should show input as disabled if isUploading is true', () => {
    const component = shallow(<FileUpload onChange={() => {}} isUploading={true} />)
    expect(component).toMatchSnapshot()
  })

})
