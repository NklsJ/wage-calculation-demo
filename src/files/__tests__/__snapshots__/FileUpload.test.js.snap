// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUpload component:  Should render correctly 1`] = `
<div>
  <h3>
    Upload CSV
  </h3>
  <input
    accept=".csv"
    disabled={false}
    multiple={true}
    onChange={[Function]}
    type="file"
  />
</div>
`;

exports[`FileUpload component:  Should show input as disabled if isUploading is true 1`] = `
<div>
  <h3>
    Upload CSV
  </h3>
  <input
    accept=".csv"
    disabled={true}
    multiple={true}
    onChange={[Function]}
    type="file"
  />
</div>
`;
